Investigate CD failure: ECS service did not stabilize

## Summary
Investigate CD failure in develop nonprod deploy run 21478676332 (ECS service failed to stabilize).

## Problem / Goal
The GitHub Actions CD workflow failed during the ECS service update step. The deploy job timed out waiting for ECS service stability.

## Failure details
- Run: https://github.com/wphillipmoore/mnemosys-core/actions/runs/21478676332
- Job: deploy
- Step: Register task definition and update service
- Error: `Waiter ServicesStable failed: Max attempts exceeded` (exit code 255)
- Service: `mnemosys-dev-api`
- Cluster: `mnemosys-nonprod`
- Task definition attempted: `arn:aws:ecs:us-east-2:959713283130:task-definition/mnemosys-development-api:75`

Relevant log excerpt:
```
Waiter ServicesStable failed: Max attempts exceeded
##[error]Process completed with exit code 255.
```

## Acceptance criteria
- Determine why ECS service failed to stabilize (task startup failure, health check failures, capacity issues, etc.).
- Identify the first failing task event or container error (CloudWatch logs, ECS events).
- Document the root cause and remediation (or rollback) steps.
- Confirm a subsequent CD run completes successfully.

## Validation / Evidence
- Link to ECS events and/or CloudWatch logs showing the failure cause.
- Successful re-run (or manual deployment) with service reaching steady state.

--- a/Dockerfile
+++ b/Dockerfile
@@ -11,7 +11,11 @@ RUN pip install --no-cache-dir -r /app/requirements.txt
 
 COPY alembic /app/alembic
 COPY alembic.ini /app/alembic.ini
+COPY pyproject.toml /app/pyproject.toml
+COPY README.md /app/README.md
+COPY LICENSE /app/LICENSE
 COPY src /app/src
+RUN pip install --no-cache-dir --no-deps /app
 COPY scripts/runtime/entrypoint.sh /app/scripts/runtime/entrypoint.sh
 
 RUN chmod +x /app/scripts/runtime/entrypoint.sh

